src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -74,10 +74,16 @@ public function getProxyFactoryCode(Definition $definition, $id)
         $methodName = 'get'.Container::camelize($id).'Service';
         $proxyClass = $this->getProxyClassName($definition);
 
+        $generatedClass = $this->generateProxyClass($definition);
```

Comment:
Aware: my initial implementation was using a `class_exists()` check, but that leads to compatibility hell later on (if I move the class, which is really internal anyway).

Since this is at dump-time only, it is not a big problem anyway. No I/O is happening due to this repeated call.